UPGRADING.md

```diff
@@ -0,0 +1,92 @@
+Before you upgrade to a new version of the Link SDK, you should review the [Releases page](https://github.com/argyle-systems/argyle-link-ios/releases). This will help you to understand any changes that you will need to make in your code as the result of an upgrade.
+
+We strongly recommend upgrading the SDK as soon as there is a new version available. Doing so will help you to provide the best Argyle Link experience in your application.
+
+---
+### Upgrade to a new SDK version
+
+`The minimum supported iOS version is iOS 12.`
+
+**Previously installed the Argyle iOS SDK using pod files**
+
+
+1. Run `pod outdated` to see if there is a newer version of the Argyle iOS SDK available. You can also refer to the [release notes](https://github.com/argyle-systems/argyle-link-ios/releases).
+
+2. If a newer version of the SDK is found, update your `Podfile` with this new version. See [Specifying pod versions](https://guides.cocoapods.org/using/the-podfile.html#specifying-pod-versions) to learn how to set a specific version in your `Podfile`. 
+
+3. Run `pod update` to update the SDK with the new version.
+
+4. Make any necessary changes as a result of the upgrade.
+
+5. Clean and rebuild your project by selecting **Product > Clean** and then **Product > Build**.
+
+---
+
+**Previously installed the Argyle iOS SDK using Carthage**
+
+1. Run `carthage outdated` to see if there is a newer version of the Argyle iOS SDK available. You can also refer to the [release notes](https://github.com/argyle-systems/argyle-link-ios/releases).
+
+2. If a newer version of the SDK is found, update your `Cartfile` with the new version. See [Version requirements](https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#version-requirement) to learn how to set a specific version in your `Cartfile`.
+
+3. Run `carthage update` to update the SDK with the new version.
+
+4. Make any necessary changes as a result of the upgrade.
+
+5. Clean and rebuild your project by selecting **Product > Clean** and then **Product > Build**.
+---
+
+**Previously installed the Argyle iOS SDK by manually downloading the SDK**
+
+1. Download the current newest SDK and replace the old one.
+
+2. Make any necessary changes as a result of the upgrade.
+
+3. Clean and rebuild your project by selecting **Product > Clean** and then **Product > Build**.
+
+---
+
+
+**To avoid issues and crashes, update your SDK to the latest version before you compile your app. If you manually downloaded your SDK framework locally, you should make sure you are using the latest SDK version.**
+
+---
+
+To verify your SDK version, you can read your `Podfile.lock` and check the `PODS: 'Insert'` value.
+
+---
+# Migrating to Link 4
+
+If you're using a previous major version of Link and would like to migrate to the new version, please update the changed configuration parameters and then follow the usual [iOS Link SDK version upgrade instruction](#upgrade-to-a-new-sdk-version) defined above.
```

Comment:
There are a couple of things changed in `Link 4`, are they gonna be described here?

Reply:
I've added the breaking change (`pluginKey` -> `linkKey`) in the subsection of this - `Migrating from Link 3`. But I guess it can be confusing and I should just drop the Link 3 part altogether.

Reply:
There are some other breaking changes i.e removal of SDK param `introSearchPlaceholder`, so yeah, either we should have a definitive list or do not do any property-per-property migration instructions.